### PR TITLE
Add minimum date option to SingleDatePicker 

### PR DESCRIPTION
**Pull request by simonemonetti**
Hi guys,
I couldn't set a specific start date to `<SingleDatePicker />` and then I realized [I was not the only one](https://github.com/airbnb/react-dates/issues/156), so I basically added a `minDate` prop that allows you to set a range of available dates starting from the minimum date provided.

It could be pretty useful in case you have to handle departure / return fields using SingleDatePicker instead of DateRangePicker.

### CODE REVIEW

File: src/components/SingleDatePicker.jsx [hunk 1]
```diff
@@ -44,7 +44,8 @@ const defaultProps = {
 
   isDayBlocked: () => false,
   disabledDays: [],
-  isOutsideRange: day => !isInclusivelyAfterDay(day, moment()),
+  minDate: moment(),
```
Comment (ljharb) [inline src/components/SingleDatePicker.jsx]: letters are free, let's name this `minimumDate` :-)

**Comment by majapw** [comment]
Wait, what exactly is this solving? Why couldn't you just do this by passing in a different `isOutsideRange` function?

Like if you want a minimumDate, couldn't you just do:
```
render() {
  const minimumDate = moment().add(1, 'week');
  return (
    <SingleDatePicker 
      ...
      isOutsideRange={(day) => day.isBefore(minimumDate)}
    />
  );
}
```

**Comment by simonemonetti** [comment]
@majapw hi Maja and thanks! You're right, that should work too. 
However I still prefer to use a different (optional) prop rather than override `isOutsideRange` (which wasn't explained very clearly on the documentation).

**Comment by majapw** [comment]
Hi @simonemonetti, I'd prefer not to add duplicate props to the components that accomplish the same thing. This seems like it is more of a documentation issue than an actual problem with the property. Let me think on how to rewrite the description of `isOutsideRange` (I've been meaning to revamp the README anyways).

**Comment by DeanPoulin** [comment]
I was just implementing this functionality myself and tried a few things and got lost. Thanks @majapw for pointing me in the right direction on how to use isOutsideRange properly. Works perfectly, and the calendar doesn't allow the user to select dates earlier than the date I've defined. 👌

**Pull request closed without merging**